substrate/client/transaction-pool/tests/integration.rs

```diff
@@ -0,0 +1,123 @@
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Testsuite of fatp integration tests.
+#[cfg(feature = "zombienet-tests")]
+pub mod zombienet;
+
+#[cfg(feature = "zombienet-tests")]
+mod zn_test {
+	use crate::zombienet::{
+		asset_hub_based_network_spec_paths::HIGH_POOL_LIMIT_FATP, default_zn_scenario_builder,
+		NetworkSpawner,
+	};
+	use txtesttool::execution_log::ExecutionLog;
+
+	// Test which sends future and ready txs from many accounts
+	// to an unlimited pool of a parachain collator based on the asset-hub-rococo runtime.
+	#[tokio::test(flavor = "multi_thread")]
+	async fn send_future_and_ready_from_many_accounts_to_parachain() {
+		let net = NetworkSpawner::from_toml_with_env_logger(HIGH_POOL_LIMIT_FATP).await.unwrap();
+
+		// Wait for the parachain collator to start block production.
+		net.wait_for_block_production("charlie").await.unwrap();
+
+		// Create future & ready txs executors.
+		let ws = net.node_rpc_uri("charlie").unwrap();
+		let future_scenario_executor = default_zn_scenario_builder(&net)
+			.with_rpc_uri(ws.clone())
+			.with_start_id(0)
+			.with_last_id(99)
+			.with_nonce_from(Some(100))
+			.with_txs_count(100)
+			.with_executor_id("future-txs-executor".to_string())
+			.build()
+			.await;
+		let ready_scenario_executor = default_zn_scenario_builder(&net)
+			.with_rpc_uri(ws)
+			.with_start_id(0)
+			.with_last_id(99)
+			.with_nonce_from(Some(0))
+			.with_txs_count(100)
+			.with_executor_id("ready-txs-executor".to_string())
+			.build()
+			.await;
+
+		// Execute transactions and fetch the execution logs.
+		let (future_logs, ready_logs) = futures::future::join(
+			future_scenario_executor.execute(),
+			ready_scenario_executor.execute(),
+		)
+		.await;
+
+		let finalized_future =
+			future_logs.values().filter_map(|default_log| default_log.finalized()).count();
+		let finalized_ready =
+			ready_logs.values().filter_map(|default_log| default_log.finalized()).count();
+
+		assert_eq!(finalized_future, 10_000);
+		assert_eq!(finalized_ready, 10_000);
+	}
+
+	// Test which sends future and ready txs from many accounts
+	// to an unlimited pool of a relaychain node based on `rococo-local` runtime.
+	#[tokio::test(flavor = "multi_thread")]
+	async fn send_future_and_ready_from_many_accounts_to_relaychain() {
+		let net = NetworkSpawner::from_toml_with_env_logger(HIGH_POOL_LIMIT_FATP).await.unwrap();
```

Comment:
Second thought on this: we could have dedicated toml for relaychain. Could be done in follow-up.

Reply:
Added it here: [0d9f118](https://github.com/paritytech/polkadot-sdk/pull/7257/commits/0d9f118e5c91af15bc2b8e0f20d903bbf124d0d6)

Reply:
Do we need to have the parachain on fatp for the relaychain test? Also, the dev accounts patching?

Reply:
No, only relay-chain is required. We don't need parachain (startup will be quicker).

Reply:
You mean we don't need the parachain entirely or just the fapt setup?

Reply:
I mean we need networks with parachains too. So all existing tomls are fine.

But for testing relaychain we could have dedicated toml file with only relaychain nodes with `devAccounts`.

Reply:
Oke. Will update soon